File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, nums: List[int], k: int, threshold: int) -> int:
        
        current_sum = 0
        count = 0

        for i in range(k):
            current_sum += nums[i]
        
        start = 0
        end = k 

        while end <= len(nums):
            if current_sum / k >= threshold:
                count+=1

            if end< len(nums):
                current_sum+= nums[end] - nums[start]
            start += 1
            end += 1

        return count
            

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] nums, int k, int threshold) {
        
        int current_sum = 0;
        int count = 0;

        for(int i = 0; i<k; i++){
            current_sum += nums[i];
        }

        int start = 0;
        int end = k;

        while(end <= nums.length){
            if(current_sum / k >= threshold){
                count++;
            }
            if(end < nums.length){
                current_sum += nums[end] - nums[start];
            }
            start++;
            end++;
        }
        return count;
    }
}